app/controllers/groups_controller.rb

```diff
@@ -44,7 +51,20 @@ def update
 
   private
 
+  def authorized?
+    if params[:token]
+      authorize!
```

Comment:
Yeah I knew something wasn't correct but couldn't find a better way. That's a good idea. I'll change this.